### PR TITLE
:art: Move find_template_match and helping fns to new file

### LINKED ISSUE #192: find_template_match should be in its own file.
Currently [find_template_match](https://github.com/fennecdjay/Gwion/blob/master/src/parse/check.c#L546-L617) is too big and takes a lot of space in `src/parse/check.c`
It would be better to move it to its own file (e.g: src/parse/func_resolve_tmpl.c` with its siblings (ensure_tmpl, check_template_args and probably a few other).
Bonus points for splitting the functions and maybe defining a struct to handle all this.

### PR DESCRIPTION
**Pull request by Pranav2612000**
Fixes #192

### CODE REVIEW

**Comment by Pranav2612000** [comment]
Are there any other functions that should be moved?

File: include/tmp_resolve.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#ifndef __TMP_RESOLVE
+#define __TMP_RESOLVE
```
Comment (fennecdjay) [inline include/tmp_resolve.h:2]: Should be __TMPL_RESOLVE

**Review by fennecdjay — COMMENTED** [review]
I think there is just the header guard to change.

**Comment by fennecdjay** [comment]
Besides the header guard thing, it looks good to me :smile:
Next thing would be to split `_find_template-match` : one function for function pointer (fptr), the other for regular template functions.
I think it deserves another issue/PR.
Are you OK to merge? (once guards are fixed)
Also, would you like to handle the splitting issue?
Thanks for the work.

File: include/tmp_resolve.h (hunk 1 above)
Comment (Pranav2612000) [reply include/tmp_resolve.h:2]: On it.

**Comment by Pranav2612000** [comment]
> Are you OK to merge? (once guards are fixed)
Yes. Let's do the splitting in another PR
> Also, would you like to handle the splitting issue?
Yes, would love to. Could you create an issue, explaining what exactly is to be done.

**Comment by fennecdjay** [comment]
> Yes, would love to. Could you create an issue, explaining what exactly is to be done.

Sure. Can I assign you already? so that people do not start working on something we know will be fixed?

**Comment by fennecdjay** [comment]
I just noticed the header you created is named `tmp_resolve`. I didn't catch it while reviewing (as a matter of fact, I'm still a newby at maintaining :smile:). IMHO, this is incorrect, as it makes us think this is some *temporary* resolving, while it concerns *templates* (yep, we miss the *function* in both case, maybe we should address that), which I generally abbreviate *tmpl* in the code.
That's another issue :smile:

**Comment by Pranav2612000** [comment]
> > Yes, would love to. Could you create an issue, explaining what exactly is to be done.
> 
> Sure. Can I assign you already? so that people do not start working on something we know will be fixed?

Yes.

**Comment by fennecdjay** [comment]
I not sure unless you comment in the issue #204 :confused:

**Comment by Pranav2612000** [comment]
> I just noticed the header you created is named `tmp_resolve`. I didn't catch it while reviewing (as a matter of fact, I'm still a newby at maintaining smile). IMHO, this is incorrect, as it makes us think this is some _temporary_ resolving, while it concerns _templates_ (yep, we miss the _function_ in both case, maybe we should address that), which I generally abbreviate _tmpl_ in the code.
> That's another issue smile

Yeah. Sorry. I should have changed that too with the header guards. I have added a new commit to the same branch which renames it.

**Pull request merged**